abort_if not treated like if statements by type checker

- Larastan Version: 0.2.4
- `--level` used: max

### Description:
Currently `abort_if` does not seem to get treated similarly to if, anything designated as nullable is still considered nullable after `abort_if`

This results in an error when querying relations on the object:
```
Cannot call method theRelation() on mixed.
```

### Workaround/example:
Current workaround for this is to use an if statement with a regular `abort`:

```php
// Doesn't pass checks
$foo = ModelName::where(['foo' => 'bar'])->first();
abort_if(!$foo, 404);

print_r($foo->theRelation());

// Passes checks
$foo = ModelName::where(['foo' => 'bar'])->first();
if (!$foo) {
  abort(404);
}

print_r($foo->theRelation());
```

On `--level=max` calling methods on mixed do trigger errors so that's probably why it occurs.

I've whipped up a repository that reproduces the error, before reading your reply fully, anyway if it's ever needed [keryja/larastan-error-examples#reproduce-abort-if](https://github.com/keryja/larastan-error-examples/commit/ec155630fd72951eb3bde59951ff4cb29940967e).

```
 22/22 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ --------------------------------------------
  Line   app/Http/Controllers/TheController.php
 ------ --------------------------------------------
  16     Cannot call method theRelation() on mixed.
 ------ --------------------------------------------

 [ERROR] Found 1 error
```

--- a/extension.neon
+++ b/extension.neon
@@ -92,3 +92,8 @@ services:
         class: NunoMaduro\Larastan\ReturnTypes\Helpers\TransExtension
         tags:
             - phpstan.broker.dynamicFunctionReturnTypeExtension
+
+    -
+        class: NunoMaduro\Larastan\Types\AbortIfFunctionTypeSpecifyingExtension
+        tags:
+            - phpstan.typeSpecifier.functionTypeSpecifyingExtension

--- a/src/Types/AbortIfFunctionTypeSpecifyingExtension.php
+++ b/src/Types/AbortIfFunctionTypeSpecifyingExtension.php
@@ -1,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\Larastan\Types;
+
+/*
+ * This file is part of Larastan.
+ *
+ * (c) Nuno Maduro <enunomaduro@gmail.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+use PHPStan\Analyser\Scope;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\TypeSpecifier;
+use PHPStan\Analyser\SpecifiedTypes;
+use PHPStan\Analyser\TypeSpecifierContext;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Analyser\TypeSpecifierAwareExtension;
+use PHPStan\Type\FunctionTypeSpecifyingExtension;
+
+final class AbortIfFunctionTypeSpecifyingExtension implements FunctionTypeSpecifyingExtension, TypeSpecifierAwareExtension
+{
+    /** @var TypeSpecifier */
+    private $typeSpecifier;
+
+    public function isFunctionSupported(
+        FunctionReflection $functionReflection,
+        FuncCall $node,
+        TypeSpecifierContext $context
+    ): bool {
+        return $functionReflection->getName() === 'abort_if' && $context->null();
+    }
+
+    public function specifyTypes(
+        FunctionReflection $functionReflection,
+        FuncCall $node,
+        Scope $scope,
+        TypeSpecifierContext $context
+    ): SpecifiedTypes {
+        return $this->typeSpecifier->specifyTypesInCondition($scope, $node->args[0]->value, TypeSpecifierContext::createFalsey());
+    }
+
+    public function setTypeSpecifier(TypeSpecifier $typeSpecifier): void
+    {
+        $this->typeSpecifier = $typeSpecifier;
+    }
+}

--- a/tests/Features/Types/AbortIf.php
+++ b/tests/Features/Types/AbortIf.php
@@ -1,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Features\Types;
+
+class AbortIf
+{
+    public function testAbortIf(?int $foo) : int
+    {
+        abort_if(! $foo, 500);
+
+        return $foo;
+    }
+
+    /**
+     * @param int|string $foo
+     * @return int
+     */
+    public function testAbortIfWithTypeCheck($foo = 5) : int
+    {
+        abort_if(is_string($foo), 500);
+
+        return $foo;
+    }
+}

